sympy/functions/elementary/integers.py

```diff
@@ -20,11 +20,24 @@ class RoundFunction(Function):
 
     @classmethod
     def eval(cls, arg):
-        from sympy import im
+        from sympy import im, fraction
         v = cls._eval_number(arg)
         if v is not None:
             return v
 
+        n, d = fraction(arg)
+        if d.is_integer:
+            if cls is floor:
+                if n == -1 and d.is_positive:
+                    return n
+                if n == 1 and d.is_negative:
+                    return -n
+            if cls is ceiling:
+                if n == 1 and d.is_positive:
+                    return n
+                if n == -1 and d.is_negative:
+                    return -n
```

Comment:
It seems odd to add this as a special case here. Really what this block is doing is:
```python
if arg.is_nonnegative and is_lt(arg, S.One):
    return S.Zero
elif arg.is_negative and is_ge(arg, S.NegativeOne):
    return S.NegativeOne
```
We could make `is_ge/is_lt` handle this by including handlers for `_eval_is_ge(Pow, ...)` and so on.

To me it would seem more useful for the codebase as a whole to have the code here depend on general purpose handlers and to improve the capabilities of the general purpose handlers for use in other parts of the codebase as well.

Reply:
Note that adding this capability to `is_ge` would immediately make it available to `Relational` as well.